### ISSUE TITLE
“React is not defined” when using fragments in MDX

### ISSUE THREAD

**Issue by joshwcomeau** [hashicorp/next-mdx-remote#127]
I've updated to use the 3.0 alpha in my codebase, and overall it works great!

I did catch one edge-case issue, around fragments.

In my MDX, I had a component like this:

```
<MultipleChoiceQuestion
  question={
    <>
      Something something
    </>
  }
/>
```

When the page is rendered, I get an error: `React is not defined`. I figured out that it was because of the fragment, <>. In 2.x, this works fine, but in 3.0 it doesn't seem to make React available directly, when parsing the elements in MDX:

![image](https://user-images.githubusercontent.com/6692932/117046290-850ef800-acde-11eb-9de5-a484785483f6.png)

The exact stack varies depending on whether it's server-rendered, or accessed from a client-side transition. Here's the server-side stack:

![image](https://user-images.githubusercontent.com/6692932/117046372-9e17a900-acde-11eb-9266-1226dad4d2c5.png)

(The code in `_document` has to do with setting styled-components up to extract styles).

I worked around this by switching <> to <Fragment>, and updating my components map to include it:

```jsx
<MDXRemote 
  components={{
    Fragment: React.Fragment,
    // All my other stuff
  }}
/>
```

(I thought I could solve a wider range of issues by making `React` available directly, but sadly, no dice. Presumably because of the distinction between React being in scope when it's run vs. React being available when the elements are created?)

Hope this info helps! 

cc @BRKalow

**Comment by BRKalow** [comment]
Thanks for the detailed report, I'll take a look at this ASAP!

**Comment by BRKalow** [comment]
This should be fixed by #128, we'll get it released shortly!

**Comment by BRKalow** [comment]
Fixed in `3.0.1` 🙏
